### PR TITLE
Add a post requestes for creating users and help requests

### PR DESCRIPTION
**Pull request by michal-kk**

### CODE REVIEW

**Comment by HelpGiver** [comment]
Maybe leave only 1 helper in request? We do not have a use case for several. Or at least do not allow adding more that 1 to the list.
Then we need to find all requests assigned to a helper. Will it be more difficult with a collection that with a single object?

File: src/main/java/com/example/helpgiver/controller/UserController.java [hunk 1]
```diff
@@ -71,9 +75,35 @@
...
+    public ResponseEntity<EntityModel<User>> updateUser(@RequestBody User user) {
+        User savedUser = userRepository.save(user);
+
+        return ResponseEntity.ok(new EntityModel<>(savedUser,
+                linkTo(methodOn(UserController.class).getUserById(savedUser.getId())).withSelfRel(),
+                linkTo(methodOn(UserController.class).getUsers()).withRel("users")));
+    }
+
+    @DeleteMapping("user")
+    public ResponseEntity<CollectionModel<Object>> deleteUser(@RequestBody User user) {
```
Comment (HelpGiver) [inline src/main/java/com/example/helpgiver/controller/UserController.java:103]: Delete can be done by ID, no need to send body just for deletion. Make ID a path variable

File: src/main/java/com/example/helpgiver/controller/HelpRequestController.java [hunk 2]
```diff
@@ -45,4 +66,58 @@
...
+        helpers.add(optionalUser.get());
+        helpRequestRepository.save(foundHelpRequest);
+
+        return ResponseEntity.ok(new EntityModel<>(foundHelpRequest,
+                linkTo(methodOn(HelpRequestController.class).getHelpRequest(foundHelpRequest.getId())).withSelfRel(),
+                linkTo(methodOn(HelpRequestController.class).getHelpRequests()).withRel("helpRequests")));
+    }
+
+    @DeleteMapping("helpRequest")
+    public ResponseEntity<CollectionModel<Object>> deleteHelpRequest(@RequestParam HelpRequest helpRequest) {
```
Comment (HelpGiver) [inline src/main/java/com/example/helpgiver/controller/HelpRequestController.java:117]: Make ID a path variable, no need to send a body for deletion.

File: src/main/java/com/example/helpgiver/controller/UserController.java (hunk 1 above)
Comment (michal-kk) [reply src/main/java/com/example/helpgiver/controller/UserController.java]: done

File: src/main/java/com/example/helpgiver/controller/HelpRequestController.java (hunk 2 above)
Comment (michal-kk) [reply src/main/java/com/example/helpgiver/controller/HelpRequestController.java]: done

**Comment by michal-kk** [comment]
> Maybe leave only 1 helper in request? We do not have a use case for several. Or at least do not allow adding more that 1 to the list.
> Then we need to find all requests assigned to a helper. Will it be more difficult with a collection that with a single object?

Yes, I've changed it to one

**Comment by shumanski** [comment]
OK

**Comment by shumanski** [comment]
Right account this time :)

**Pull request merged**